### PR TITLE
Add date index to order tables for speeding up order list query.

### LINKED ISSUE #35465: [HPOS] Modify index for `type_status` to `type_status_date` to accommodate created date in the same index.
Many queries, such as the query for calculating unique year-months in `Orders\ListTable` class (function `months_filter`) needs to filter on date_created along with type_status. This issue scopes modifying this index to add `date_created_gmt` field to `type_status`.

Additionally, currently the query in `months_filter` function, is not filtering by order type. This issue also scopes adding another clause for order type.

### PR DESCRIPTION
**Pull request by vedanshujain**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This adds the equivalent index of wp_posts' `type_status_date`, the main impact of this index would be on the order list page for large sites. Additionally, it also connects WC_Install's create table method to HPOS when it is enabled.

Closes #35465 

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Set HPOS to authoritative, or data sync to enabled, or both. (So that HPOS tables are registered and created).
2. Testing this needs to trigger the update flow, the easiest option is to update the WooCommerce version to a lower one, like so: 
`wp option update woocommerce_version 7.9.0`
3. Refresh any WooCommerce admin page, so that update is triggered. 
4. Verify that woocommerce version was updated, by running `wp option get woocommerce_version`, it should be greater than 7.9.0.
5. Check that the index was created as expected, by running ` wp db query "show index from wp_wc_orders where Key_name like 'type_status_date'"`, it should show 3 rows for each index in query like this:
```
+--------------+------------+------------------+--------------+------------------+-----------+-------------+----------+--------+------+------------+---------+---------------+
| Table        | Non_unique | Key_name         | Seq_in_index | Column_name      | Collation | Cardinality | Sub_part | Packed | Null | Index_type | Comment | Index_comment |
+--------------+------------+------------------+--------------+------------------+-----------+-------------+----------+--------+------+------------+---------+---------------+
| wp_wc_orders |          1 | type_status_date |            1 | type             | A         |           3 |     NULL | NULL   | YES  | BTREE      |         |               |
| wp_wc_orders |          1 | type_status_date |            2 | status           | A         |          10 |     NULL | NULL   | YES  | BTREE      |         |               |
| wp_wc_orders |          1 | type_status_date |            3 | date_created_gmt | A         |         309 |     NULL | NULL   | YES  | BTREE      |         |               |
+--------------+------------+------------------+--------------+------------------+-----------+-------------+----------+--------+------+------------+---------+---------------+
```

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/includes/class-wc-install.php [hunk 1]
```diff
@@ -1162,6 +1162,11 @@ private static function get_schema() {
 
 		$product_attributes_lookup_table_creation_sql = wc_get_container()->get( DataRegenerator::class )->get_table_creation_sql();
 
+		$feature_controller = wc_get_container()->get( FeaturesController::class );
+		$hpos_enabled =
```
Comment (barryhughes) [inline plugins/woocommerce/includes/class-wc-install.php:1166]: Not a request to change anything, more asking out of curiosity. At this stage, with HPOS on the verge of being marked stable, might we not simply create the HPOS tables as standard?

**Review by barryhughes — APPROVED** [review]
Left a non-blocking question (it's not a request for changes, though). Looks good and tests well (key was established, order list table functions as before). Thanks!

File: plugins/woocommerce/includes/class-wc-install.php (hunk 1 above)
Comment (vedanshujain) [reply plugins/woocommerce/includes/class-wc-install.php:1166]: Actually, that's not a bad idea. I think we can do in preparation for a larger move. Would you like to create an issue for this?

Comment (barryhughes) [reply plugins/woocommerce/includes/class-wc-install.php:1166]: [Created](https://github.com/woocommerce/woocommerce/issues/39826) (and added to the project board) ✅

**Pull request merged**